src/dataflow-types/src/client/controller.rs

```diff
@@ -56,6 +60,71 @@ pub struct OrchestratorConfig {
     pub storage_addr: String,
 }
 
+fn deserialize_memory_limit<'de, D>(deserializer: D) -> Result<Option<ByteSize>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let maybe_string = Option::<String>::deserialize(deserializer)?;
+    match maybe_string {
+        None => Ok(None),
+        Some(string) => match ByteSize::from_str(&string) {
+            Ok(bs) => Ok(Some(bs)),
+            Err(_e) => {
+                use serde::de::Error;
+                Err(D::Error::invalid_value(
+                    serde::de::Unexpected::Str(&string),
+                    &"valid size in bytes",
+                ))
+            }
+        },
+    }
+}
```

Comment:
Feels a little weird to have this here. Can we instead make that `MemoryLimit` struct that used to exist a newtype over `ByteSize`, and then add this custom deserialize impl on that type?

Reply:
SGTM, I have no real preference

Reply:
Or, we could have our own `ByteSize` equivalent? It doesn't seem we're using too much of the functionality offered by the crate.

Reply:
@antiguru the main functionality we are using is `ByteSize::from_str` which handles the MB/MiB/GB/GiB/etc. suffixes, which we could write ourselves, but I can't see a good reason to do so rather than just taking the crate.